expat/tests/fuzz/parse_fuzzer.cc

```diff
@@ -0,0 +1,47 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "expat.h"
+
+#include <functional>
+#include <string>
+
+const char* kEncoding =
+#if defined(ENCODING_UTF_16)
+"UTF-16"
+#elif defined(ENCODING_UTF_8)
+"UTF-8"
+#elif defined(ENCODING_ISO_8859_1)
+"ISO-8859-1"
+#elif defined(ENCODING_US_ASCII)
+"US-ASCII"
+#elif defined(ENCODING_UTF_16BE)
+"UTF-16BE"
+#elif defined(ENCODING_UTF_16LE)
+"UTF-16LE"
+#else
+#error Encoding type is not specified.
+#endif
+;
+
+// Entry point for LibFuzzer.
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+  std::string input(reinterpret_cast<const char*>(data), size);
+  auto hash_salt = std::hash<std::string>()(input);
```

Comment:
> What's the idea behind deriving hash salt from input?

Have a reproducible behavior, avoiding randomness as the comment in the code states a few lines below.
Fuzzing works much better when a same input has a reproducible output...

> Is use of auto in this line the only reason why C++98 is not good enough for this code?

There is `std::hash<std::string>()` as well

Reply:
> > Is use of auto in this line the only reason why C++98 is not good enough for this code?
> 
> There is `std::hash<std::string>()` as well

I see! 


> Have a reproducible behavior, avoiding randomness as the comment in the code states a few lines below.
> Fuzzing works much better when a same input has a reproducible output...

No question about need for determinism, yes. I was just wondering why a fixed number would not do well here. By now, I think it might help to have some variance on the salt as well to not be blind to potential issues that might be masked by one single hardcoded salt value. So hashing is not a bad idea, maybe! Since we have the luxury of having a C++98 compatible `siphash.h` as part of Expat already, we could make use of `siphash.h` here, instead. Would be great to leverage SipHash to get back to C++98.

Reply:
@yevgenypats did you see this^^?